### ISSUE TITLE
How do I create a graph that has metadata but does not use labeled vertices?

### ISSUE THREAD

**Issue by Krastanov** [JuliaGraphs/MetaGraphsNext.jl#37]
With the old `MetaGraphs` it was easy to make a graph where indexing was still done with integers, not with symbolic labels. How do I do this with `MetaGraphsNext`.

E.g., how do I do `g = MetaGraph(Graph()); add_vertex!(g)`? (and maybe then followed by `set_prop!(g,1,:key,:value)`)

**Comment by bramtayl** [comment]
How about

```
using Graphs
using MetaGraphsNext
meta_graph = MetaGraph(Graph(), Label = Int, VertexData = Pair{Symbol, Symbol})
meta_graph[1] = :key => :value
```

This will work, you will just get a warning, because it's easy to confuse integer vertex labels with integer vertex codes.

**Comment by Krastanov** [comment]
This seems like a neat workaround! I am still curious though, does this not involve an extra lookup step that needs to map an integer vertex label to an integer vertex code? Is there something I can do in order to avoid this computational expense if all I need is vertex/edge metadata, without the need for custom labels?

**Comment by bramtayl** [comment]
Yes, there is an extra lookup step. Do you plan on deleting vertices? If so, I think it's probably worth it. Graphs.jl reassigns vertex codes when a vertex is deleted.

**Comment by Krastanov** [comment]
Thanks! This was very helpful and it answers all my questions. I would leave this issue opened and rename it to focus on adding this to the documentation. Please close it if it is a duplicate.

**Comment by bramtayl** [comment]
Ok, feel free to make a PR to update the docs!

**Comment by gdalle** [comment]
Hi @Krastanov, would you like to contribute by documenting this workaround?

**Comment by gdalle** [comment]
@bramtayl if we decide to forbid integer labels for good, what happens to this method?

**Comment by bramtayl** [comment]
Yeah this wouldn't work anymore...part of why I'm conflicted

**Comment by gdalle** [comment]
One alternative could be to provide a custom integer type with the package, which would not subtype `Integer`, so that people can use it as label. Something like
```julia
struct IntegerLabel{T<:Integer}
    n::T
end
const IL = IntegerLabel
```
but it seems a bit overkill

**Comment by gdalle** [comment]
#53 removes the warning for integer labels, as long as it's well-documented I think we're okay
